docs/std-str.md

```diff
@@ -1,127 +1,109 @@
+| `//str.sub("this is the old string", "old string", "new sentence")` | `"this is the new sentence"` |
+| `//str.sub("just another sentence", "string", "stuff")` | `"just another sentence"` |
 
-## split
+## `split(s <: string, delimiter <: string) <: array of string`
 
-`//.str.split(s, delimiter)` returns a list of string which are splitted from the string `s`
-based on a given delimiter. It takes two arguments, `s` is the base string, and
-`delimiter` is a string that defines the boundaries for each split. It returns
-an array of strings.
+Split the string `s` based on the provided `delimiter`. It returns an array of string
+which are splitted from the string `s`.
 
-#### examples
+Usage:
 
-| expression | equals |
+| example | equals |
 |:-|:-|
-| `//.str.split("deliberately adding spaces to demonstrate the split function", " ")` | `["deliberately", "adding", "spaces", "to", "demonstrate", "the", "split", "function"]` |
-| `//.str.split("this is just a random sentence", "random stuff")` | `["this is just a random sentence"]` |
+| `//str.split("deliberately adding spaces to demonstrate the split function", " ")` | `["deliberately", "adding", "spaces", "to", "demonstrate", "the", "split", "function"]` |
+| `//str.split("this is just a random sentence", "random stuff")` | `["this is just a random sentence"]` |
 
-## lower
+## `lower(s <: string) <: string`
 
-`//.str.lower(string)` returns a string where each letter is converted to its lowercase
-form. It takes one argument which is a string that you want to convert. It
-returns the converted string.
+Returns the string `s` except all of the character is converted to its lowercase form.
 
-#### examples
+Usage:
 
-| expression | equals |
+| example | equals |
 |:-|:-|
-| `//.str.lower("HeLLo ThErE")` | `"hello there"` |
-| `//.str.lower("GENERAL KENOBI WHAT A SURPRISE")` | `"general kenobi what a surprise"` |
-| `//.str.lower("123")` | `"123"` |
+| `//str.lower("HeLLo ThErE")` | `"hello there"` |
+| `//str.lower("GENERAL KENOBI WHAT A SURPRISE")` | `"general kenobi what a surprise"` |
+| `//str.lower("123")` | `"123"` |
 
-## upper
+## `upper(s <: string) <: string`
 
-`//.str.upper(string)` returns a string where each letter is converted to its uppercase
-form. It takes one argument which is a string that you want to convert. It
-returns the converted string.
+Returns the string `s` except all of the character is converted to its uppercase form.
 
-#### examples
+Usage:
 
-| expression | equals |
+| example | equals |
 |:-|:-|
-| `//.str.upper("HeLLo ThErE")` | `"HELLO THERE"` |
-| `//.str.upper("did you ever hear the tragedy of darth plagueis the wise")` | `"DID YOU EVER HEAR THE TRAGEDY OF DARTH PLAGUEIS THE WISE"` |
-| `//.str.upper("321")` | `"321"` |
+| `//str.upper("HeLLo ThErE")` | `"HELLO THERE"` |
+| `//str.upper("did you ever hear the tragedy of darth plagueis the wise")` | `"DID YOU EVER HEAR THE TRAGEDY OF DARTH PLAGUEIS THE WISE"` |
```

Comment:
Sounds terrifying! Should there be a space in plagueis?

Reply:
haha meant to be a placeholder but I forgot to replace it again. And nope!